homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,179 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_ADDRRESS = 'gateway_address'
+
+REQUIREMENTS = ['zhong_hong_hvac==1.0.8']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST):
+    cv.string,
+    vol.Optional(CONF_PORT, default=9999):
+    vol.Coerce(int),
+    vol.Optional(CONF_GATEWAY_ADDRRESS, default=1):
+    vol.Coerce(int),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZhongHong HVAC platform."""
+    from zhong_hong_hvac.hub import ZhongHongGateway
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    gw_addr = config.get(CONF_GATEWAY_ADDRRESS)
+    hub = ZhongHongGateway(host, port, gw_addr)
+
+    def startup(event):
+        """Add devices to HA and start hub socket."""
+        try:
+            devices = [
+                ZhongHongClimate(hub, addr_out, addr_in)
+                for (addr_out, addr_in) in hub.discovery_ac()
+            ]
+        except Exception as exc:
+            _LOGGER.error("ZhongHong controller is not ready", exc_info=exc)
+            raise PlatformNotReady
+        add_devices(devices)
+        hub.start_listen()
+        hub.query_all_status()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, startup)
+
+
+class ZhongHongClimate(ClimateDevice):
+    """Representation of a ZhongHong controller support HVAC."""
+
+    def __init__(self, hub, addr_out, addr_in):
+        """Set up the ZhongHong climate devices."""
+        from zhong_hong_hvac.hvac import HVAC
+        self._device = HVAC(hub, addr_out, addr_in)
+        self._hub = hub
+        self._device.register_update_callback(self._after_update)
+
+    def _after_update(self, climate):
+        """Callback to update state."""
+        _LOGGER.debug("async update ha state")
+        self.schedule_update_ha_state()
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self.unique_id
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the HVAC."""
+        return "zhong_hong_hvac_{}_{}".format(self._device.addr_out,
+                                              self._device.addr_in)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+                | SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._device.current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._device.operation_list
```

Comment:
This list should only contain states found in the climate base component.

Reply:
DONE